app/controllers/Application.scala

```diff
@@ -1,12 +1,194 @@
 package controllers
 
+import play.api.Play.current
 import play.api._
+import play.api.libs.json.Json.JsValueWrapper
 import play.api.mvc._
+import play.api.db._
+import play.twirl.api.Html
+import play.api.libs.json._
+
+import scala.collection.mutable.ArrayBuffer
 
 class Application extends Controller {
 
   def index = Action {
-    Ok  }
+    Ok
+  }
+
+
+  def listAgents = Action {
+    var jsonBuffer = ArrayBuffer.empty[JsValue]
+    val conn = DB.getConnection()
+    try {
+      val stmt = conn.createStatement
+      val rs = stmt.executeQuery("SELECT id, first_name, last_name, user_name from AGENT")
+      while (rs.next()) {
+        val json: JsValue = Json.obj(
+          "id" -> rs.getInt("ID"),
+          "firstName" -> rs.getString("first_name"),
+          "lastName" -> rs.getString("last_name"),
+          "userName" -> rs.getString("user_name")
+        )
+        jsonBuffer += json
+      }
+      rs.close()
+      stmt.close()
+    }
+    finally {
+      conn.close()
+    }
+    val j = JsArray(jsonBuffer)
+    Ok(j)
+  }
+
+
+  def validateAgent() = Action {
+
+    Ok
+  }
+
+
+  def addAgent() = Action {
+
+    Ok
+  }
+
+
+  //TODO we want to take some parameters for this call
+  // - Office location
+  // - Agent ID for tickets assigned to that agent
+  // - can these be
+  def listTickets = Action {
```

Comment:
I’ve made the changes. Maybe rebase we do after the merge to get master as well. Please review again.

 

From: nafg [mailto:notifications@github.com] 
Sent: Wednesday, June 17, 2015 4:42 AM
To: smith4312/TAG-Ticket
Cc: smith4312
Subject: Re: [TAG-Ticket] Db1 (#20)

 

In app/controllers/Application.scala <https://github.com/smith4312/TAG-Ticket/pull/20#discussion_r32605459> :

> +
> +    Ok
> +  }
> +
> +
> +  def addAgent() = Action {
> +
> +    Ok
> +  }
> +
> +
> +  //TODO we want to take some parameters for this call
> +  // - Office location
> +  // - Agent ID for tickets assigned to that agent
> +  // - can these be
> +  def listTickets = Action {

I think at some point we need to factor out the redundancy between all these controller actions (and get rid of vars for bonus points), but I'm not sure if it should wait... (depends on the עולם's readiness...)

—
Reply to this email directly or view it on GitHub <https://github.com/smith4312/TAG-Ticket/pull/20/files#r32605459> .  <https://github.com/notifications/beacon/ALe6bA85-5rfur4AUjleaYpXDMB_g1XQks5oUSpQgaJpZM4FEcWB.gif>